### PR TITLE
Implement extra cursors on macOS

### PR DESCRIPTION
**Pull request by JonnyPtn**
Addresses #1401

The move cursor doesn't look quite as I'd expect (It's basically just a white arrow) but I can't see any better looking alternatives.

### CODE REVIEW

**Comment by mantognini** [comment]
SFMLBuildMaster: Build this please.

This looks good but haven't actually tested it. I'll try to do it this week. Thanks for the PR!

**Comment by mantognini** [comment]
I guess the documentation should be updated as well, specifically these lines:

https://github.com/JonnyPtn/SFML/blob/76f3adbd978e8955d24dcc787339563edc307b90/include/SFML/Window/Cursor.hpp#L65-L67 

What about replacing *no* with *maybe (1)* and adding a note along those lines: *(1) those cursors are usually available on macOS \<insert minimal version\> and newer but without any guarantee.* (Or phrased completely differently...)

**Comment by JonnyPtn** [comment]
Ah yes, Sorry I missed the docs, will update now. 

Not really sure what to put for the supported version, can't find any documentation about it, and looking at other projects doesn't yield any useful results (glfw has an open (issue)[https://github.com/glfw/glfw/issues/427], and as far as I can tell SDL just uses the closed hand cursor for these diagonal resizes (?)

File: src/SFML/Window/OSX/CursorImpl.mm [hunk 1]
```diff
@@ -80,6 +90,14 @@
...
         case Cursor::Cross:           m_cursor = [NSCursor crosshairCursor];           break;
         case Cursor::NotAllowed:      m_cursor = [NSCursor operationNotAllowedCursor]; break;
+            
+        // These cursor types are undocumented, may not be available on some platforms
+        case Cursor::SizeBottomLeftTopRight:
+            return loadFromSelector(@selector(_windowResizeNorthEastSouthWestCursor), m_cursor);
+        case Cursor::SizeTopLeftBottomRight:
+            return loadFromSelector(@selector(_windowResizeNorthWestSouthEastCursor), m_cursor);
+        case Cursor::SizeAll:
+            return loadFromSelector(@selector(_moveCursor), m_cursor);
```
Comment (JonnyPtn) [inline src/SFML/Window/OSX/CursorImpl.mm]: Will returning here cause issues as it doesn't hit `[m_cursor retain]`?

**Comment by JonnyPtn** [comment]
I've added the Help and Wait cursors, and updated documentation as best I can.

You can use this test code to cycle through the new cursors with Space:
```
#include <SFML/Graphics.hpp>

int main(int, char const**)
{
    // Load cursors
    std::vector<sf::Cursor> cursors(5);
    
    cursors[0].loadFromSystem(sf::Cursor::SizeBottomLeftTopRight);
    cursors[1].loadFromSystem(sf::Cursor::SizeTopLeftBottomRight);
    cursors[2].loadFromSystem(sf::Cursor::SizeAll);
    cursors[3].loadFromSystem(sf::Cursor::Wait);
    cursors[4].loadFromSystem(sf::Cursor::Help);
    
    int currentCursor = -1;
    
    // Create the main window
    sf::RenderWindow window(sf::VideoMode(800, 600), "SFML window");

    // Start the game loop
    while (window.isOpen())
    {
        // Process events
        sf::Event event;
        while (window.pollEvent(event))
        {
            // Close window: exit
            if (event.type == sf::Event::Closed)
            {
                window.close();
            }

            // Escape pressed: exit
            if (event.type == sf::Event::KeyPressed && event.key.code == sf::Keyboard::Escape)
            {
                window.close();
            }
            if (event.type == sf::Event::KeyPressed && event.key.code == sf::Keyboard::Space)
            {
                currentCursor++;
                if (currentCursor >= cursors.size())
                    currentCursor = 0;
                window.setMouseCursor(cursors[currentCursor]);
            }
        }

        // Clear screen 
        window.clear();
    }

    return EXIT_SUCCESS;
}

```

**Comment by mantognini** [comment]
Yes, most likely. Good catch!

**Comment by mantognini** [comment]
OK, that's good. But we should make it clear that Apple can remove those in
the future.

I'll see to test this tomorrow or the day after. :)

**Comment by mantognini** [comment]
Superseded by #1425.

**Pull request closed without merging**